src/Symfony/Component/Validator/Constraints/ExactlyValidator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Hamza Amrouche <hamza.simperfit@gmail.com>
+ */
+class ExactlyValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (null === $value) {
+            return;
+        }
+
+        if (!$constraint instanceof Exactly) {
+            throw new UnexpectedTypeException($constraint, Exactly::class);
+        }
+
+        if (!is_iterable($value)) {
+            throw new UnexpectedValueException($value, 'array or Traversable');
+        }
+
+        $totalIterations = \count($value) * \count($constraint->constraints);
+
+        $validator = $this->context->getValidator()->inContext($this->context);
```

Comment:
don't validate in the existing context. This will cause issues if other validators have run and have added some violations themselves.
You need to run it on a dedicated context (which is supported precisely for this kind of features).